### PR TITLE
Partial fix for Agent

### PR DESCRIPTION
**Pull request by jablonskim**

### CODE REVIEW

File: src/endian.cc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include <endian.h>
+
+#include <boost/endian/conversion.hpp>
+
+uint64_t htobe64(uint64_t host_64bits) {
+    return boost::endian::native_to_big(host_64bits);
```
Comment (mclapinski) [inline src/endian.cc]: Actually I'd prefer if you put this directly in agent code (instead of htobe64).

**Review by mclapinski — CHANGES_REQUESTED** [review]

**Review by mclapinski — APPROVED** [review]

File: inc/sys/wait.h [hunk 2]
```diff
@@ -1 +1,16 @@
-#pragma once
+#ifndef __WAIT_H__
```
Comment (kaweue) [inline inc/sys/wait.h]: Maybe we can use here  _SYS_WAIT_H like in original one. It is more unique.

**Review by kaweue — CHANGES_REQUESTED** [review]

File: inc/sys/wait.h (hunk 2 above)
Comment (krdln) [reply inc/sys/wait.h]: I'd go with `__SYS_WAIT_H__`

**Review by kaweue — APPROVED** [review]

File: src/posix_time.cc [hunk 3]
```diff
@@ -1,9 +1,14 @@
 #include <posix_time.h>
+
 #include <sstream>
 #include <iomanip>
+#include <boost/algorithm/string/replace.hpp>
 
-char* strptime(const char* str,const char* format, struct tm* tm) {
-    std::istringstream input(str);
+char* strptime(const char* str, const char* format, struct tm* tm) {
+    std::string replaced(str);
```
Comment (krdln) [inline src/posix_time.cc]: I think we wanted to do `replaced(format)`

File: src/posix_time.cc [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
+    // strptime supports %F and std::get_time doesn't not, so we replace it
+    boost::algorithm::replace_all(replaced, "%F", "%Y-%m-%d");
+    std::istringstream input(replaced);
     input.imbue(std::locale(setlocale(LC_ALL, nullptr)));
     input >> std::get_time(tm, format);
```
Comment (krdln) [inline src/posix_time.cc]: We're still passing the old `format` here

**Review by krdln — CHANGES_REQUESTED** [review]

File: src/posix_time.cc [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
 #include <iomanip>
+#include <boost/algorithm/string/replace.hpp>
 
-char* strptime(const char* str,const char* format, struct tm* tm) {
-    std::istringstream input(str);
+char* strptime(const char* str, const char* format, struct tm* tm) {
+    std::string replaced(str);
+    // strptime supports %F and std::get_time doesn't not, so we replace it
+    boost::algorithm::replace_all(replaced, "%F", "%Y-%m-%d");
+    std::istringstream input(replaced);
```
Comment (krdln) [inline src/posix_time.cc]: ... and `input(str)`

**Review by krdln — CHANGES_REQUESTED** [review]

**Review by krdln — APPROVED** [review]

**Pull request merged**